karma.conf.js

```diff
@@ -0,0 +1,133 @@
+// Karma configuration
+// Generated on Fri Sep 25 2020 11:02:54 GMT+0200 (Central European Summer Time)
+
+const { join: joinPath } = require( 'path' );
+
+const basePath = process.cwd();
+const coverageDir = joinPath( basePath, 'coverage' );
+
+module.exports = function(config) {
+	config.set({
+
+		// base path that will be used to resolve all patterns (eg. files, exclude)
+		basePath,
+
+
+		// frameworks to use
+		// available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+		frameworks: ['mocha', 'chai', 'sinon'],
+
+
+		// list of files / patterns to load in the browser
+		files: [
+			'tests/**/*.js'
+		],
+
+
+		// preprocess matching files before serving them to the browser
+		// available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
+		preprocessors: {
+			'tests/**/*.js': [ 'webpack' ]
+		},
+
+
+		webpack: {
+			mode: 'development',
+			devtool: 'inline-source-map',
+
+			module: {
+				rules: [
+					{
+						test: /\.js$/,
+						loader: 'babel-loader',
+						exclude: /node_modules/,
+						query: {
+							compact: false,
+							presets: [ '@babel/preset-env' ]
+						}
+					},
+					{
+						test: /\.js$/,
+						loader: 'istanbul-instrumenter-loader',
+						include: /src/,
+						exclude: [
+							/node_modules/
+						],
+						query: {
+							esModules: true
+						}
+					}
+				]
+			}
+		},
+
+		webpackMiddleware: {
+			noInfo: true,
+			stats: 'minimal'
+		},
+
+
+		// test results reporter to use
+		// possible values: 'dots', 'progress'
+		// available reporters: https://npmjs.org/browse/keyword/karma-reporter
+		reporters: [
+			'mocha',
+			'coverage'
+		],
+
+
+		coverageReporter: {
+			reporters: [
+				// Prints a table after tests result.
+				{
+					type: 'text'
+				},
+				// Generates HTML tables with the results.
+				{
+					dir: coverageDir,
+					type: 'html'
+				},
+				// Generates "lcov.info" file. It's used by external code coverage services.
+				{
+					type: 'lcovonly',
+					dir: coverageDir
+				}
+			]
+		},
+
+
+		// web server port
+		port: 9876,
+
+
+		// enable / disable colors in the output (reporters and logs)
+		colors: true,
+
+
+		// level of logging
+		// possible values: config.LOG_DISABLE || config.LOG_ERROR || config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
+		logLevel: config.LOG_INFO,
+
+
+		// enable / disable watching file and executing tests whenever any file changes
+		autoWatch: false,
+
+
+		// start these browsers
+		// available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
+		browsers: ['ChromeHeadless'],
```

Comment:
Just my 5 cents, shouldn't we test it on other browsers too - Firefox, Safari, Edge, IE11 since we plan to use this paclage with integrations which should work on all those browsers? :thinking: Or the code is really simple (for now :stuck\_out\_tongue:) that it shouldn't make a difference?

Reply:
Well, it makes sense... Also in CI (currently only Chrome is set).

Reply:
The package is written in ES6 because it should be used as a dependency and transpiled by "host". I've added tests for ChromeHeadless because it's blazing fast and we don't really care about compatibility here as it should be assured by the main lib. I would keep that way for simplicity, WDYT? cc @Comandeer what do you think about that approach?

Reply:
It sounds reasonable, but at the same time every of our integrations has its own bundling approach (mainly because every framework approaches it differently). We'd need to test if deps are really bundled with the rest of the integration.

However testing should be done here. I don't see any clear benefits of delegating it to other packages.

Reply:
Thanks, @Comandeer, looks like I lose!